driver-async/src/main/com/mongodb/async/client/MongoClients.java

```diff
@@ -155,6 +156,8 @@ public static MongoClient create(final ConnectionString connectionString, final
         if (connectionString.getStreamType() != null) {
             if (connectionString.getStreamType().toLowerCase().equals("netty")) {
                 builder.streamFactoryFactory(NettyStreamFactoryFactory.builder().build());
+            } else if (connectionString.getStreamType().toLowerCase().equals("nio2")) {
+                builder.streamFactoryFactory(new AsynchronousSocketChannelStreamFactoryFactory());
```

Comment:
Will do I triggered it by running the driver-async tests.  Jenkins runs the driver tests with netty - so it was missed.